src/de/gurkenlabs/litiengine/graphics/animation/AsepriteHandler.java

```diff
@@ -0,0 +1,412 @@
+   * @throws ExportAnimationException if the export fails
+   */
+  private static String createJson(SpritesheetResource spritesheetResource) throws ExportAnimationException {
+    Spritesheet spritesheet = Resources.spritesheets().load(spritesheetResource);
+    assert spritesheet != null;
+    int[] keyframes = Resources.spritesheets().getCustomKeyFrameDurations(spritesheet);
+    Frames[] frames = new Frames[keyframes.length];
+    
+    if (frames.length != spritesheet.getTotalNumberOfSprites()) {
+      throw new ExportAnimationException("Different dimensions of keyframes and sprites in spritesheet");
+    }
+    
+    // Build the frames object in the json
+    int numCol = spritesheet.getColumns();
+    int numRows = spritesheet.getRows();
+    int frameWidth = spritesheet.getSpriteWidth();
+    int frameHeight = spritesheet.getSpriteHeight();
+    
+    for (int i = 0; i < numRows; i++) {
+      for (int j = 0; j < numCol; j++) {
+        final int row = i;
+        final int col = j;
+        Map<String, Integer> frame = new HashMap<String, Integer>() {{
+          put("x", (0 + col * frameWidth));
+          put("y", (0 + row * frameHeight));
+          put("w", frameWidth);
+          put("h", frameHeight);
+        }};
+        Map<String, Integer> spriteSourceSize = new HashMap<String, Integer>() {{
+          put("x", 0);
+          put("y", 0);
+          put("w", frameWidth);
+          put("h", frameHeight);
+        }};
+        Map<String, Integer> sourceSize = new HashMap<String, Integer>() {{
+          put("w", frameWidth);
+          put("h", frameHeight);
+        }};
+        int duration = keyframes[i + j];
+        String index = String.valueOf(i + j);
+        frames[i + j] = new Frames("frame " + index,
+                frame,
+                false,
+                false,
+                spriteSourceSize,
+                sourceSize,
+                duration);
+      }
+    }
+    
+    // Build the meta object in the json
+    int spritesheetWidth = frameWidth * numCol;
+    int spritesheetHeight = frameHeight * numRows;
+    Map<String, Integer> size = new HashMap<String, Integer>() {{
+      put("w", spritesheetWidth);
+      put("h", spritesheetHeight);
+    }};
+    String spritesheetName = spritesheet.getName();
+    Layer[] layers = {new Layer("Layer", 255, "normal")};
+    Meta meta = new Meta("http://www.aseprite.org/",
+            "1.2.16.3-x64",
+            spritesheetName,
+            "RGBA8888", size, "1", layers);
+    
+    // Create the json as string
+    Gson gson = new GsonBuilder().setPrettyPrinting().create();
+    StringBuilder sb = new StringBuilder();
+    
+    sb.append("{ \"frames\": {\n");
+    for (int i = 0; i < frames.length; i++) {
+      String json = gson.toJson(frames[i]);
+      sb.append(" \"" + frames[i].name + "\": ").append(json).append(",\n");
+    }
+    sb.append(" },\n");
+    String json = gson.toJson(meta);
+    sb.append("\"meta\":").append(json).append("\n}");
```

Comment:
I'm not sure if it's appropriate to address this in this PR, because this more within the scope of https://github.com/gurkenlabs/litiengine/issues/331, but Gson isn't used appropriately here.

- Creating a new Gson instance is expensive. Gson is thread safe, so we can use a singleton instance for the entirety of the engine and expose the instance to any games developed for it.

- We should be creating a custom gson serializer for the animation, and register it using `GsonBuilder#registerTypeAdapter()`. We would have to also allow games to add their own gson serializers before `Game#init()` is called.

What do you think @nightm4re94, @steffen-wilke? Should this be addressed in this PR, or leave it be and address it in another PR for #331?

Reply:
I personally think it's out of scope for this PR, and I think  it should be addressed in another PR. Otherwise this PR will probably get too bloated.

Reply:
Thank you for the feedback. #321 stated _"Maybe even an export could be useful."_  which made us add it, but maybe it fits better in a separate PR.

One other interesting aspect is that Aseprite does not allow .json import, only export.

Reply:
I agree that the PR is very feature-rich already and shouldn't be stuffed with the proposed changes from #331.